Source/OxyPlot/Series/HeatMapSeries.cs

```diff
@@ -170,6 +185,58 @@ public void Invalidate()
         {
             this.image = null;
         }
+        
+        /// <summary>
+        /// Gets the clipping rectangle, transposed if the X acis is vertically orientated
+        /// </summary>
+        protected new OxyRect GetClippingRect()
+        {
+            double minX = Math.Min(this.XAxis.ScreenMin.X, this.XAxis.ScreenMax.X);
+            double minY = Math.Min(this.YAxis.ScreenMin.Y, this.YAxis.ScreenMax.Y);
+            double maxX = Math.Max(this.XAxis.ScreenMin.X, this.XAxis.ScreenMax.X);
+            double maxY = Math.Max(this.YAxis.ScreenMin.Y, this.YAxis.ScreenMax.Y);
+
+            if (XAxis.IsVertical())
+                return new OxyRect(minY, minX, maxY - minY, maxX - minX);
+            else
+                return new OxyRect(minX, minY, maxX - minX, maxY - minY);
+        }
+
+        /// <summary>
+        /// Transposes the ScreenPoint if the X axis is vertically orientated
+        /// </summary>
+        private ScreenPoint Orientate(ScreenPoint point)
+        {
+            if (XAxis.IsVertical())
+            {
+                point = new ScreenPoint(point.Y, point.X);
+            }
+            return point;
+        }
+
+        /// <summary>
+        /// Transforms DataSpace coords to orientated ScreenSpace coors
+        /// </summary>
+        public new ScreenPoint Transform(double x, double y)
+        {
+            return Orientate(base.Transform(x, y));
+        }
+
+        /// <summary>
+        /// Transforms DataSpace coords to orientated ScreenSpace coors
+        /// </summary>
+        public new ScreenPoint Transform(DataPoint p)
+        {
+            return Orientate(base.Transform(p));
+        }
+
+        /// <summary>
+        /// Transforms orientated ScreenSpace coords to DataSpace coors
+        /// </summary>
+        public new DataPoint InverseTransform(ScreenPoint point)
+        {
+            return base.InverseTransform(Orientate(point));
+        }
```

Comment:
I would be interested to hear some comments on these methods - I can't say I'm especially fond of hiding underlying methods like this, but intercepting these does, I think, make sense, and would be applicable to other XYAxisSeries types (line and scatter, for example) which also struggle with Transposing, as per #535. I am wondering if these 'overloads' could be moved into XYAxisSeries, or some intermediary (by way of example, LineSeries can be 'fixed' by just pasting these shadow methods into LineSeries.cs, though there may be some marker details that need attention). I should highlight that while HeatMapSeries does have to work around this fix itself, it shouldn't be a common issue.